src/modules/Elsa.Workflows.Management/Providers/AssemblyPathActivityProvider.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Elsa.Workflows.Core.Contracts;
+using Elsa.Workflows.Core.Models;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace Elsa.Workflows.Management.Providers;
+
+/// <summary>
+/// Provides activity descriptors based on a list of NON-ELSA activity types finded into the same directory of the assembly.
+/// </summary>
+public class AssemblyPathActivityProvider : IActivityProvider
+{
+    private readonly IActivityDescriber _activityDescriber;
+
+    public AssemblyPathActivityProvider(IActivityDescriber activityDescriber)
+    {
+        _activityDescriber = activityDescriber;
+    }
+
+    /// <inheritdoc />
+    public async ValueTask<IEnumerable<ActivityDescriptor>> GetDescriptorsAsync(CancellationToken cancellationToken = default)
+    {
+        var activityTypes = GetActivityTypes();
+        var descriptors = await DescribeActivityTypesAsync(activityTypes, cancellationToken).ToListAsync(cancellationToken);
+        return descriptors;
+    }
+
+    private async IAsyncEnumerable<ActivityDescriptor> DescribeActivityTypesAsync(IEnumerable<Type> activityTypes, [EnumeratorCancellation] CancellationToken cancellationToken)
+    {
+        foreach (var activityType in activityTypes)
+        {
+            var descriptor = await _activityDescriber.DescribeActivityAsync(activityType, cancellationToken);
+            yield return descriptor;
+        }
+    }
+
+    private IEnumerable<Type> GetActivityTypes()
+    {
+        var currentAssembly = Assembly.GetExecutingAssembly();
+
+        // Get the directory of the current assembly
+        var path = Path.GetDirectoryName(currentAssembly.Location);
+
+        // Get all the assembly files in the directory
+        var assemblyFiles = System.IO.Directory.GetFiles(path, "*.dll")
+            .Where(f => !f.Split('\\').ToList().Last().StartsWith("Elsa"));
```

Comment:
Does this work for Linux-based systems? Would it be better to split on e.g. `Path. PathSeparator`?